pandas/core/window/rolling.py

```diff
@@ -1889,6 +1891,14 @@ def _validate(self):
             else:
                 self._win_freq_i8 = freq.nanos
 
+                unit = dtype_to_unit(self._on.dtype)
+                if unit == "us":
+                    self._win_freq_i8 /= 1e3
```

Comment:
should this use floordiv?

Reply:
you might be better off doing this higher up instead of doing `freq.nanos` above (which could raise for very large freq anyway) doing `Timedelta(freq).as_unit(unit)._value`

Reply:
Thanks for the review and the suggestion. Didn't get it quite well in which situation could make an issue? Could I have the suggested code with the direct modification in a comment, please?

Reply:
IIUC, it would be like `self._win_freq_i8 = Timedelta(freq).as_unit(unit)._value`, and then you don't need the if/else checks.

Reply:
Great! That's a util function we were originally searching for :) I have updated the code